go/vt/worker/split_clone.go

```diff
@@ -912,25 +912,32 @@ func mergeOrSingle(readers []ResultReader, td *tabletmanagerdatapb.TableDefiniti
 	return sourceReader, nil
 }
 
+func closeReaders(ctx context.Context, readers []ResultReader) {
+	for _, reader := range readers {
+		if reader != nil {
+			reader.Close(ctx)
+		}
+	}
+}
+
 func (scw *SplitCloneWorker) getSourceResultReader(ctx context.Context, td *tabletmanagerdatapb.TableDefinition, state StatusWorkerState, chunk chunk, txID int64) (ResultReader, error) {
 	sourceReaders := make([]ResultReader, len(scw.sourceShards))
-	var readers []ResultReader
-	defer func() {
```

Comment:
Do we need to be concerned around panics?

Reply:
If you're worried about readers being nil: go's range construct treats `nil` as an empty list. So, there won't be a panic.

Reply:
I thought the concern was that an uncaught panic might cause the readers to not be closed.